build/networking/README_DOCKER.md

```diff
@@ -77,226 +75,122 @@ container up and running, and should see the container prompt like below:
+sudo ./root/scripts/set_hugepages.sh
 ```
 
-2) On your host, create two Ubuntu demo VM images with accompanying cloud-init images:
+### Export all environment variables to sudo user
 
 ```
-ipdk createvms
+alias sudo='sudo PATH="$PATH" HOME="$HOME" LD_LIBRARY_PATH="$LD_LIBRARY_PATH" SDE_INSTALL="$SDE_INSTALL"'
 ```
 
-3) Those two created VM images can at any moment be started with:
+### Run the infrap4d daemon
 
 ```
-ipdk startvms
+cd $IPDK_RECIPE
+sudo ./install/sbin/infrap4d
 ```
 
-4) Create a forwarding pipeline program by compiling and package the vSwitch consumable
-pipeline binary package by using the pipeline builder:
+By default, infrap4d runs in detached mode. If you want to run infrap4d in attached mode, use the --nodetach option.
 
-```
-export OUTPUT_DIR=/root/examples/simple_l3
-ipdk execute --- p4c --arch psa --target dpdk --output $OUTPUT_DIR/pipe --p4runtime-files $OUTPUT_DIR/p4Info.txt --bf-rt-schema $OUTPUT_DIR/bf-rt.json --context $OUTPUT_DIR/pipe/context.json $OUTPUT_DIR/simple_l3.p4
+  - All infrap4d logs are by default logged under /var/log/stratum.
+  - All P4SDE logs are logged in p4_driver.log under $IPDK_RECIPE.
+  - All OVS logs are logged under /tmp/ovs-vswitchd.log.
 
-ipdk execute /root/examples/simple_l3 --- ovs_pipeline_builder --p4c_conf_file=simple_l3.conf --bf_pipeline_config_binary_file=simple_l3.pb.bin
-```
+As your infrap4d should be up and running, you could see the infrap4d  process with the ps -ef | grep infrap4d command.
 
-5) Add the created pipeline binary package to the running IPDK container P4-OVS switch:
+### Run a sample program
 
-```
-ipdk execute --- ovs-p4ctl set-pipe br0 /root/examples/simple_l3/simple_l3.pb.bin /root/examples/simple_l3/p4Info.txt
-```
+Make sure that all the environmental variables are set correctly.
 
-6) Add pipeline table rules:
+#### Create 2 TAP ports
 
 ```
-ipdk execute --- ovs-p4ctl add-entry br0 ingress.ipv4_host "hdr.ipv4.dst_addr=1.1.1.1,action=ingress.send(0)"
+cd $IPDK_RECIPE
 
-ipdk execute --- ovs-p4ctl add-entry br0 ingress.ipv4_host "hdr.ipv4.dst_addr=2.2.2.2,action=ingress.send(1)"
+sudo ./install/bin/gnmi-ctl set "device:virtual-device,name:TAP0,pipeline-name:pipe,mempool-name:MEMPOOL0,mtu:1500,port-type:TAP"
+sudo ./install/bin/gnmi-ctl set "device:virtual-device,name:TAP1,pipeline-name:pipe,mempool-name:MEMPOOL0,mtu:1500,port-type:TAP"
+ifconfig TAP0 up
+ifconfig TAP1 up
 ```
 
-From this step you can connect to the test VMs and do some ping tests as described below!
-
-### Connect to the test VMs serial consoles
+Note: See gnmi-ctl Readme for more information on the gnmi-ctl utility.
 
-You will need two login windows, one for each VM:
+### Create P4 artifacts
 
 ```
-telnet localhost 6551
+export OUTPUT_DIR=/root/examples/simple_l3
 ```
 
-And in another window:
+#### Generate the artifacts using the p4c compiler installed in the previous step
```

Comment:
`previous step`?? I dont see any mention of p4c compiler before this statement.

If missed out, we need to provide more details to user on how and where to get the p4compiler.